src/Compilers/CSharp/Portable/Errors/LazyUnmanagedCallersOnlyMethodCalledDiagnosticInfo.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class LazyUnmanagedCallersOnlyMethodCalledDiagnosticInfo : DiagnosticInfo
```

Comment:
>LazyUnmanagedCallersOnlyMethodCalledDiagnosticInfo [](start = 26, length = 50)

Do we still need this class?

Reply:
Obsolete is bound early, and still uses a lazy diagnostic. In order to make sure that we truly don't hit any cycles, I'm going to leave this to mirror how obsolete works.

---
In reply to: [483713350](https://github.com/dotnet/roslyn/pull/47311#discussion_r483713350) [](ancestors = 483713350)